moving_modules.md

```diff
@@ -0,0 +1,358 @@
+
+# Changes
+
+## puppet repo
+
+Add a new puppet setting `vendormoduledir` defaulting to:
+
+    /opt/puppetlabs/puppet/vendor_modules # On *nix
+    C:\Program Files\Puppet Labs\Puppet\puppet\vendor_modules # On Windows
+
+The autoloader will include the `vendormoduledir` in its search path with least
+precedence.
+
+It will be possible to set `vendormoduledir = ` so that the autoloader ignores vendored
+modules. For example, if you want to manage all module dependencies via Puppetfile.
+
+All of the modules in the [`External`](#external) section below will be removed from the
+puppet repo and extracted into modules on the forge.
+
+## puppet-agent package
+
+Several of the external modules will be added to the puppet-agent package, e.g.
+`host`, and they will be installed into the `vendormoduledir`.
+
+## puppet gem
+
+When installing puppet as a gem, external modules will not be present. But
+they can be installed using a Puppetfile or `puppet module install` commands
+such as in the provisioning step of a vagrant workflow.
+
+## Forge
+
+We will be publishing extracted modules to the forge using the PDK and following
+modules best practices.
+
+# Resource Type Taxonomy
+
+## Internal
+
+The following types are internal to Puppet and will remain as is:
+
+    component
+    schedule
+    stage
+    whit
+
+## Core
+
+The following types will be left in Puppet for now. The `file`, `user`, and `group` types are needed to apply settings catalogs. The `filebucket`, `resources`, and `tidy` types know too much about puppet internals to be extracted. The `notify` type is used extensively in puppet rspec tests as it is the most basic (providerless) type. The `package` and `service` types have multiple providers for each type, which makes removal more difficult. We may extract them at a later time (TBD).
+
+    exec
+    file
+    filebucket
+    group
+    notify
+    package
+    resources
+    service
+    tidy
+    user
+
+## External
+
+The following types and providers will be extracted from Puppet. A subset (details TBD below) will be added back to puppet-agent during packaging.
+
+Each top-level path below specifies the name of the module, e.g. `augeas`, and the files contained within each module. The modules will be installed in a new directory visible to Puppet's autoloader, so catalog compilation and application will just work without additional configuration. Puppet will prefer modules in the modulepath and pluginsync'ed lib directory over the packaged modules, so that newer versions of modules can fix bugs in packaged modules.
+
+    Path                                           Comments
+
+    <vendor_modules>/                              (*nix) /opt/puppetlabs/puppet/vendor_modules/
```

Comment:
Should this also get added to https://github.com/puppetlabs/puppet-specifications/blob/master/file_paths.md ?

Reply:
Good point, I will update that.